src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/StartupCode/StartupCodeHelpers.cs

```diff
@@ -63,43 +57,24 @@ internal static unsafe void InitializeCommandLineArgs(int argc, byte** argv)
             Environment.SetCommandLineArgs(args);
         }
 
-        private static unsafe IntPtr[] CreateModuleManagers()
+        private static unsafe IntPtr[] CreateModuleManagers(IntPtr moduleHeaders, int count)
         {
-            IntPtr* modulesSectionStart;
-            IntPtr* modulesSectionEnd;
-            GetModulesSection(out modulesSectionStart, out modulesSectionEnd);
-
             // Count the number of modules so we can allocate an array to hold the ModuleManager objects.
             // At this stage of startup, complex collection classes will not work.
             int moduleCount = 0;
-            for (IntPtr* current = modulesSectionStart; current < modulesSectionEnd; current++)
+            for (int i = 0; i < count; i++)
             {
-                IntPtr* moduleHeaderStart = (IntPtr*)*current;
-
-                if (moduleHeaderStart != null)
-                {
+                // TODO: Why are we getting the zero pointers in the list?
```

Comment:
Ok, if C++ runtime startup has the same problem ... it should be good enough for us too.